gbpservice/nfp/orchestrator/config_drivers/heat_driver.py

```diff
@@ -445,6 +484,69 @@ def _get_all_heat_resource_keys(self, template_resource_dict,
                 resource_keys.append(key)
         return resource_keys
 
+    def _create_firewall_template(self, auth_token, service_details, stack_template):
```

Comment:
Check, update does lot of db gets and rest apis, while new function takes from the passed variables. Since the update function is used in other functions or could possibly be used, hence separated out the path for optimized create and other paths.
Actually, all the optimized path with be like this separate functions, so one request please compare two functions.